src/contactlistviewdelegate.cpp

```diff
@@ -546,7 +546,7 @@ QPixmap ContactListViewDelegate::Private::avatarIcon(const QModelIndex &index)
     if(av.isNull() && useDefaultAvatar_)
         av = IconsetFactory::iconPixmap("psi/default_avatar");
 
-    return AvatarFactory::roundedAvatar(av, avatarRadius_, avSize);
+    return AvatarFactory::roundedAvatar(av, avatarRadius_ * devicePixelRatio(contactList), avSize * devicePixelRatio(contactList));
```

Comment:
You don't need this. avatar size and its radius are already have have physical metrics. not pixels.

Reply:
Then something is broken, as this change alone fixes avatars on roster list. Removing it makes them pixelated again.

Reply:
Qt works fine, everything is scaled properly in KDE apps. I'm using GNU/Linux with X11, BTW.

pointToPixel doesn't seem as something that should be used here, btw. It's using the logical DPI of the desktop, which breaks multiscreen and probably makes some other not always correct assumptions as well. I can't see how it can work well without specifying devicePixelRatio for bitmaps. See https://doc.qt.io/qt-5/highdpi.html

Reply:
Also, everything is *always consistently* pixelated here without those patches. I don't know anything about hidpi handling of Windows, but on GNU/Linux it sure isn't right.

Reply:
devicePixelRatio often doesn't work properly according to mutiple tests on different OS. I have no idea why, but I had reports where ratio was 1 on hidpi. 
In general it's a good idea to compute manually from dpi. If you know better algo, please propose. But for now pointToPixel usually works fine. But accroding to the article we probably also have to use Qt::AA_DisableHighDpiScaling.

I have no idea how usually multiple monitors with totally different dpi problem solved.

Reply:
Usually a user has to set the device pixel ratio. KDE Plasma has a UI for that. Under X11, it uses one factor for everything (unless you use experimental environment vars that aren't exposed in the GUI), but under Wayland each screen has its own scaling factor. Using DPI is only a good idea for supporting legacy stuff, it's almost never right with proper, modern config.

Reply:
Hm then can we just drop all the custom scaling and enable it globally?

Reply:
Well, it is enabled globally and generally works very well here. Fonts and UI elements are nicely scaled. However, for pixmaps you need to somehow tell Qt that it shouldn't scale them up because they're already high resolution and meant to be shown as-is, and when you create the pixmaps by yourself you need to ensure that you don't create them in logical, but physical size, by multiplying requested sized by device pixel ratio. Hence my patches.

I suspect that, with proper Qt scaling enabled, pointToPixel could be simply dropped. At least for avatars, no idea if it has any other use through the codebase and how would it relate to Qt scaling :)

Reply:
Ah, of course it is enabled globally here on KDE Plasma. It should be on macOS too, but on Windows it's probably disabled, and therefore Psi uses its custom scaling based on PSI_HIDPI macro, which is why you are seeing completely different results on Windows :) I would give Qt::AA_EnableHighDpiScaling a go; automatic scaling could be clunky years ago, but it's pretty mature there days.